### PR TITLE
Perceiver

### PR DESCRIPTION
**Pull request by anthonynorth**
The `an_perceiver` module implements the DeepMind Perceiver in tensorflow keras. It has been used in a classification experiment to classify knee laterality in the AOI AKOA MRI knee slices; test accuracy achieved was 99.45%; details for this experiment can be found in the README.md in [AOI AKOA knee laterality classification experiment](https://github.com/anthonynorth/PatternFlow/blob/topic-recognition/recognition/an_perceiver/README.md#aoi-akoa-knee-laterality-classification-experiment).

## Usage

Example usage of the Perceiver model:
```python
from perceiver import Perceiver
# perceiver classifier for aoi akoa knee laterality
perciever = Perceiver(
    num_blocks=2,
    num_self_attends_per_block=4,
    num_cross_heads=1,
    num_self_attend_heads=8,
    latent_dim=128,
    latent_channels=256,
    num_freq_bands=32,
    num_classes=2,
)
```

End-to-end example usage can be found in the experiment driver: `an_perceiver/experiment.py`, which runs the AOI AKOA laterality classification experiment (accepts _some_ configuration options via named command-line arguments).

The configuration used for the _best_ model (runnable on an NVIDIA RTX 2080) is:
```shell
python experiment.py \
    --epochs 10 \
    --image_dims 64 64 \
    --num_blocks 2 \
    --num_self_attends_per_block 4 \
    --latent_dim 128 \
    --latent_channels 256 \
    --num_freq_bands 32 \
```

## Dependencies
Python 3.9.6

Packages:

- tensorflow 2.6.0
- tensorflow_datasets 4.4.0
- tensorflow-addons 0.14.0

Install dependencies with `pip install -r requirements.txt`.

## Unit tests
Unit tests are contained within `_test.py` files. All tests are passing & are runnable with `pytest` (not a dependency).

  
##

Anthony North
Student \#: 46139896

### CODE REVIEW

**Comment by SiyuLiu0329** [comment]
Difficulty Hard

Functionality: OK

Code
- Structure:  OK

README 
- Content: 

PR Target OK

General Comments
- Commit messages: OK
- Excellent

**Comment by shakes76** [comment]
## TF/Torch Usage
Adequate use and implementation
## Good Practice (Design/Commenting)
Good spacing and comments
Header blocks
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage 
Module present
No Data leakage
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
Good Description and Comments 
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description very good
